File: playService/assets/src/common/images.js
function image(url) {
  const img = new Image
  img.src = url
  return img
}

const iconClose = image('/assets/ic-close.svg')
const iconGamerlist = image('/assets/ic-gamerlist.svg')
const iconReplayAnimation = image('/assets/ic-replay-animation.svg')

const players = [
  image('/assets/player1.svg'),
  image('/assets/player2.svg'),
  image('/assets/player3.svg'),
  image('/assets/player4.svg'),
  image('/assets/player5.svg'),
  image('/assets/player6.svg')
]

function player(index) {
  return players[index % 6]
}

// https://materialdesignicons.com/icon/flag-variant-outline
// const target = image('/assets/target.svg')

const actions = {
  MoveForward: image('/assets/action-move-forward.svg'),
  MoveBackward: image('/assets/action-move-backward.svg'),
// StepRight : image('/assets/action_StepRight.svg'),
// StepLeft : image('/assets/action_StepLeft.svg'),
  MoveTwiceForward: image('/assets/action-move-forward-twice.svg'),
  TurnRight: image('/assets/action-turn-right-60.svg'),
  TurnLeft: image('/assets/action-turn-left-60.svg'),
  UTurn: image('/assets/action-turn-left-180.svg'),
  Sleep: image('/assets/action-sleep.svg'),
}

function action(name) {
  return actions[name]
}

module.exports = {
  player, action,
  iconClose, iconGamerlist, iconReplayAnimation
}

File: frontend/src/main/scala/roborace/frontend/pages/components/Images.scala
package roborace.frontend.pages.components

import entities._

object Images {
  val logo: String = "/assets/logo.svg"

  val tile: String = "/assets/tile.svg"

  val target: String         = "/assets/target.svg"
  val targetInactive: String = "/assets/target-inactive.svg"

  def wall(wallDirection: WallDirection): String = wallDirection match {
    case Down      => "/assets/wall-down.svg"
    case UpRight   => "/assets/wall-up-right.svg"
    case DownRight => "/assets/wall-down-right.svg"
  }

  def player(index: Int): String = s"/assets/player${index + 1}.svg"

  def playerStart(index: Int): String = s"/assets/player${index + 1}-start.svg"

  def instructionIcon(inst: Instruction): String = inst match {
    case MoveForward      => "/assets/action-move-forward.svg"
    case MoveBackward     => "/assets/action-move-backward.svg"
    case StepRight        => ???
    case StepLeft         => ???
    case MoveTwiceForward => "/assets/action-move-forward-twice.svg"
    case TurnRight        => "/assets/action-turn-right-60.svg"
    case TurnLeft         => "/assets/action-turn-left-60.svg"
    case UTurn            => "/assets/action-turn-left-180.svg"
    case Sleep            => "/assets/action-sleep.svg"
  }

  def trapStun          = "/assets/trap-stun.svg"
  def trapTurnRight     = "/assets/trap-turn-right.svg"
  def trapTurnLeft      = "/assets/trap-turn-left.svg"
  def trapPushUp        = "/assets/trap-push-up.svg"
  def trapPushUpRight   = "/assets/trap-push-up-right.svg"
  def trapPushDownRight = "/assets/trap-push-down-right.svg"
  def trapPushDown      = "/assets/trap-push-down.svg"
  def trapPushDownLeft  = "/assets/trap-push-down-left.svg"
  def trapPushUpLeft    = "/assets/trap-push-up-left.svg"

  def trap(t: Trap): String = t match {
    case _: StunTrap            => trapStun
    case _: TurnRightTrap       => trapTurnRight
    case _: TurnLeftTrap        => trapTurnLeft
    case PushTrap(_, Up)        => trapPushUp
    case PushTrap(_, UpRight)   => trapPushUpRight
    case PushTrap(_, DownRight) => trapPushDownRight
    case PushTrap(_, Down)      => trapPushDown
    case PushTrap(_, DownLeft)  => trapPushDownLeft
    case PushTrap(_, UpLeft)    => trapPushUpLeft
  }

  val greetingBackground = "/assets/background.svg"
}
